lua/hbac/command/subcommands.lua

```diff
@@ -0,0 +1,45 @@
+local autocommands = require("hbac.autocommands")
+local hbac_config = require("hbac.setup").opts
+local state = require("hbac.state")
+local utils = require("hbac.utils")
+
+local M = {}
+
+M.close_unpinned = function()
+	local curbufnr = vim.api.nvim_get_current_buf()
+	local buflist = utils.get_buffers()
+	for _, bufnr in ipairs(buflist) do
+		local windows_with_buf = vim.fn.win_findbuf(bufnr)
+		if bufnr ~= curbufnr and not state.is_pinned(bufnr) and #windows_with_buf == 0 then
+			hbac_config.close_command(bufnr)
+		end
```

Comment:
Not a fan of the conjunct triplet here but maybe that's nitpicky

Reply:
It's either this or extracting it to a function with a series of ifs 🤷🏻